internal/corazawaf/transaction.go

```diff
@@ -748,8 +749,95 @@ func (tx *Transaction) ProcessRequestHeaders() *types.Interruption {
 	return tx.interruption
 }
 
-func (tx *Transaction) RequestBodyWriter() io.Writer {
-	return tx.RequestBodyBuffer
+type Lenger interface {
+	Len() int
+}
+
+func (tx *Transaction) WriteRequestBody(b []byte) (*types.Interruption, int, error) {
```

Comment:
While this method isn't as problematic as the previous one but it requires that we read the entire body on beforehand and pass it.